src/Kiota.Builder/Extensions/OpenApiSchemaExtensions.cs

```diff
@@ -113,14 +107,42 @@ public static bool IsInherited(this OpenApiSchema? schema)
         if (schema is null || !schema.IsExclusiveUnion(0)) return null;
         var result = new OpenApiSchema(schema);
         result.OneOf.Clear();
-        foreach (var subSchema in schema.OneOf)
+        result.TryAddProperties(schema.OneOf.SelectMany(static x => x.Properties));
+        return result;
+    }
+
+    internal static OpenApiSchema? MergeSingleInclusiveUnionInheritanceOrIntersectionSchemaEntries(this OpenApiSchema? schema)
+    {
+        if (schema is not null
```

Comment:
Can you please add unit tests calling those methods specifically?

Reply:
Added some unit tests in `OpenApiSchemaExtensionsTests` to cover these new extension methods.